h2o-genmodel/src/main/java/hex/genmodel/AbstractMojoWriter.java

```diff
@@ -0,0 +1,247 @@
+    writekv("algorithm", model.algoFullName());
+    writekv("endianness", ByteOrder.nativeOrder());
+    writekv("category", model.getModelCategory());
+    writekv("uuid", model.uuid());
+    writekv("supervised", model.isSupervised());
+    writekv("n_features", model.nfeatures());
+    writekv("n_classes", model.nclasses());
+    writekv("n_columns", model.columnNames().length);
+    writekv("n_domains", n_categoricals);
+    writekv("balance_classes", model.balanceClasses());
+    writekv("default_threshold", model.defaultThreshold());
+    writekv("prior_class_distrib", Arrays.toString(model.priorClassDist()));
+    writekv("model_class_distrib", Arrays.toString(model.modelClassDist()));
+    writekv("timestamp", model.timestamp());
+  }
+
+  /**
+   * Create the model.ini file containing 3 sections: [info], [columns] and [domains]. For example:
+   * [info]
+   * algo = Random Forest
+   * n_trees = 100
+   * n_columns = 25
+   * n_domains = 3
+   * ...
+   * h2o_version = 3.9.10.0
+   * <p>
+   * [columns]
+   * col1
+   * col2
+   * ...
+   * <p>
+   * [domains]
+   * 5: 13 d000.txt
+   * 6: 7 d001.txt
+   * 12: 124 d002.txt
+   * <p>
+   * Here the [info] section lists general model information; [columns] is the list of all column names in the input
+   * dataframe; and [domains] section maps column numbers (for categorical features) to their domain definition files
+   * together with the number of categories to be read from that file.
+   */
+  private void writeModelInfo() throws IOException {
+    startWritingTextFile("model.ini");
+    writeln("[info]");
+    for (Map.Entry<String, String> kv : lkv.entrySet()) {
+      writeln(kv.getKey() + " = " + kv.getValue());
+    }
+
+    writeln("\n[columns]");
+    for (String name : model.columnNames()) {
+      writeln(name);
+    }
+
+    writeln("\n[domains]");
+    String format = "%d: %d d%03d.txt";
+    int domIndex = 0;
+    String[][] domains = model.scoringDomains();
+    for (int colIndex = 0; colIndex < domains.length; colIndex++) {
+      if (domains[colIndex] != null)
+        writeln(String.format(format, colIndex, domains[colIndex].length, domIndex++));
+    }
+    finishWritingTextFile();
+  }
+
+  /**
+   * Create files containing domain definitions for each categorical column.
+   */
+  private void writeDomains() throws IOException {
+    int domIndex = 0;
+    for (String[] domain : model.scoringDomains()) {
+      if (domain == null) continue;
+      startWritingTextFile(String.format("domains/d%03d.txt", domIndex++));
+      for (String category : domain) {
+        writeln(category.replaceAll("\n", "\\n"));  // replace newlines with "\n" escape sequences
+      }
+      finishWritingTextFile();
+    }
+  }
+
+  private static byte[] toBytes(Object value) {
+    return bytesOf(String.valueOf(value));
+  }
+
+  private static byte[] bytesOf(CharSequence str) {
+    return str.toString().getBytes(Charset.forName("UTF-8"));
```

Comment:
If the only usage of `bytesOf()` method is on line 239 ^^^ , than this method might just be deleted.The `toString` operation is useless, and the argument type could have been String instead of  CharSequence.

Reply:
makes sense